src/Allure.XUnit/AllureXunitHelper.cs

```diff
@@ -1,185 +1,284 @@
-            var testResult = new TestResult
+        public static void StartStep(string name)
+        {
+            var stepResult = new StepResult
             {
-                uuid = _testResultGuid,
-                name = _test.DisplayName,
-                historyId = _test.DisplayName,
-                fullName = _test.DisplayName,
-                labels = new List<Label>
-                {
-                    Label.Thread(),
-                    Label.Host(),
-                    Label.TestClass(_test.TestMethod.TestClass.Class.Name),
-                    Label.TestMethod(_test.DisplayName),
-                    Label.Package(_test.TestMethod.TestClass.Class.Name)
-                }
+                name = name,
+                stage = Stage.running,
+                start = DateTimeOffset.Now.ToUnixTimeMilliseconds()
             };
-            AllureLifecycle.StartTestCase(_containerGuid, testResult);
+            var parent = Steps.Value.LastOrDefault();
+            if (parent == null)
+            {
+                parent = TestResult.Value;
+            }
+
+            parent.steps ??= new();
+            parent.steps.Add(stepResult);
+            Steps.Value.Add(stepResult);
+            Log($"Started Step: {name}");
         }
 
-        public void StopAll(RunSummary runSummary, ExceptionAggregator aggregator, bool isWrappedIntoStep = false)
+        public static void PassStep()
         {
-            StopTestCase(runSummary, aggregator);
+            PassStep(Steps.Value.Last());
+        }
 
-            StopTestContainer();
+        public static void PassStep(ExecutableItem step)
+        {
+            Steps.Value.Remove(step);
+            step.stage = Stage.finished;
+            step.stop = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+            step.status = Status.passed;
+            Log("Passed");
         }
 
-        private void StopTestCase(RunSummary runSummary, ExceptionAggregator aggregator)
+        public static void FailStep()
         {
-            UpdateTestDataFromAttributes();
-            if (StatusDetails.ContainsKey(_test.UniqueID))
-                AllureLifecycle.UpdateTestCase(x => x.statusDetails = StatusDetails[_test.UniqueID]);
+            FailStep(Steps.Value.Last());
+        }
 
-            AllureLifecycle.StopTestCase(testCase => testCase.status = GeTestStatus(runSummary));
-            AllureLifecycle.WriteTestCase(_testResultGuid);
+        public static void FailStep(ExecutableItem step)
+        {
+            Steps.Value.Remove(step);
+            step.stage = Stage.finished;
+            step.stop = DateTimeOffset.Now.ToUnixTimeMilliseconds();
+            step.status = Status.failed;
+            Log("Failed");
         }
 
-        private void StopTestContainer()
+        public static async Task AddAttachment(string name, string type, byte[] content, string fileExtension)
```

Comment:
About moving method - ok.
About another AsyncLocal field, I would disagree. We already have static AsyncLocal TestResult in AllureXunitHelper. For simplicity, it can be internal. Seems, that this will be enouth. More AsyncLocal field will raise complexity.

Reply:
I've just read another comments and will try suggested design